x-pack/test/plugin_functional/plugins/resolver_test/public/applications/resolver_test/index.tsx

```diff
@@ -4,119 +4,95 @@
-  embeddable: Promise<IEmbeddable | undefined>
+  coreStart: CoreStart,
+  parameters: AppMountParameters,
+  resolverPluginSetup: ResolverPluginSetup
 ) {
   /**
    * The application DOM node should take all available space.
    */
-  element.style.display = 'flex';
-  element.style.flexGrow = '1';
+  parameters.element.style.display = 'flex';
+  parameters.element.style.flexGrow = '1';
 
   ReactDOM.render(
-    <I18nProvider>
-      <AppRoot embeddable={embeddable} />
-    </I18nProvider>,
-    element
+    <AppRoot
+      coreStart={coreStart}
+      parameters={parameters}
+      resolverPluginSetup={resolverPluginSetup}
+    />,
+    parameters.element
   );
 
   return () => {
-    ReactDOM.unmountComponentAtNode(element);
+    ReactDOM.unmountComponentAtNode(parameters.element);
   };
 }
 
-const AppRoot = styled(
-  React.memo(
-    ({
-      embeddable: embeddablePromise,
-      className,
-    }: {
-      /**
-       * A promise which resolves to the Resolver embeddable.
-       */
-      embeddable: Promise<IEmbeddable | undefined>;
-      /**
-       * A `className` string provided by `styled`
-       */
-      className?: string;
-    }) => {
-      /**
-       * This state holds the reference to the embeddable, once resolved.
-       */
-      const [embeddable, setEmbeddable] = React.useState<IEmbeddable | undefined>(undefined);
-      /**
-       * This state holds the reference to the DOM node that will contain the embeddable.
-       */
-      const [renderTarget, setRenderTarget] = React.useState<HTMLDivElement | null>(null);
-
-      /**
-       * Keep component state with the Resolver embeddable.
-       *
-       * If the reference to the embeddablePromise changes, we ignore the stale promise.
-       */
-      useEffect(() => {
-        /**
-         * A promise rejection function that will prevent a stale embeddable promise from being resolved
-         * as the current eembeddable.
-         *
-         * If the embeddablePromise itself changes before the old one is resolved, we cancel and restart this effect.
-         */
-        let cleanUp;
-
-        const cleanupPromise = new Promise<never>((_resolve, reject) => {
-          cleanUp = reject;
-        });
-
-        /**
-         * Either set the embeddable in state, or cancel and restart this process.
-         */
-        Promise.race([cleanupPromise, embeddablePromise]).then((value) => {
-          setEmbeddable(value);
-        });
+const AppRoot = React.memo(
+  ({
+    coreStart,
+    parameters,
+    resolverPluginSetup,
+  }: {
+    coreStart: CoreStart;
+    parameters: AppMountParameters;
+    resolverPluginSetup: ResolverPluginSetup;
+  }) => {
+    const {
+      Provider,
+      storeFactory,
+      ResolverWithoutProviders,
+      mocks: {
+        dataAccessLayer: { noAncestorsTwoChildren },
+      },
+    } = resolverPluginSetup;
+    const dataAccessLayer: DataAccessLayer = useMemo(
+      () => noAncestorsTwoChildren().dataAccessLayer,
```

Comment:
Just out of curiosity, how would you see us expanding this to support multiple `dataAccessLayers`. Some kind of flag or config?

Reply:
great question +1